internal/binder/binder.go

```diff
@@ -124,9 +125,32 @@ func bindSourceFile(file *ast.SourceFile, options *core.SourceFileAffectingCompi
 		b.bind(file.AsNode())
 		file.SymbolCount = b.symbolCount
 		file.ClassifiableNames = b.classifiableNames
+		b.delayedBindJSDocTypedefTag()
 	})
 }
 
+// top-level typedef binding is delayed because it changes based on whether `module.exports = x` is bound
+func (b *Binder) delayedBindJSDocTypedefTag() {
+	if b.delayedTypeAliases == nil {
+		return
+	}
+	if b.file.Symbol != nil {
+		if exportEq := b.file.Symbol.Exports[ast.InternalSymbolNameExportEquals]; exportEq != nil && b.file.CommonJSModuleIndicator != nil {
+			for _, node := range b.delayedTypeAliases {
+				b.declareSymbol(ast.GetSymbolTable(&exportEq.Exports), exportEq /*parent*/, node, ast.SymbolFlagsTypeAlias, ast.SymbolFlagsTypeAliasExcludes)
```

Comment:
This works for one level of indirection to a local namespace-y thing, but what about an aliased namespacey thing? Eg
```js
const mod = require("./mod.js")
/**
* @typedef {string} T
*/
module.exports = mod
```
or, abusing a ts construct to do it locally,
```ts
namespace ns {
  export namespace inner {}
}
import mod = ns.inner
module.exports = mod
```
`mod`'s symbol will be an alias, and this will just patch the typedefs onto the alias symbol, and not the alias symbol's ultimate target, which, dollars to donuts, means it's going to effectively go missing without extra lookup logic in the checker, similar to what we used to have. 🥹

Reply:
Even simpler, this also reproduces the problem.

```js
function f() { }
/** @typedef {string} T */
module.expotrs = f
```